art/attacks/evasion/pixel_threshold.py

```diff
@@ -282,18 +313,19 @@ def callback_fn(x, convergence=None):  # pylint: disable=R1710,W0613
                     predict_fn,
                     maxfun=max(1, 400 // len(bounds)) * len(bounds) * 100,
                     callback=callback_fn,
-                    iterations=max_iter,
+                    iterations=self.max_iter,
                 )
-            except Exception as exception:  # pylint: disable=W0703
-                logger.info(exception)
+            except Exception as exception:
+                if self.verbose_es:
+                    print(exception)
```

Comment:
The exception's stack trace should be printed anyways.
```suggestion
            except Exception as exception:  # pylint: disable=W0703
                if self.verbose_es:
                    logger.exception("CMAEvolutionStrategy optimize failed.")
```

Reply:
Actually Most of the time, it would be informative and not a real exception handling.
CMA-ES docs recommend raising an exception to early stop optimisation. 
To do this, an exception is raised at line 288 in callback_fn, this line will print the same exception.

However, if you think, this workaround can be done better. I am open to suggestions.

Reply:
Ok, I see. How about defining a custom exception, that would also avoid the requirement for `# pylint: disable=W0703`? How about adding
 ```
class CMAEarlyStoppingException(Exception):
    """Raised when CMA is stopping early after successful optimisation."""
    pass
```
to this file?

and changing here to 
```suggestion
            except CMAEarlyStoppingException as e:
                if self.verbose_es:
                    logger.info(e.message)
```

Reply:
And change line 297 from
```
raise Exception("Attack Completed :) Earlier than expected")
```
to
```
raise CMAEarlyStoppingException("Attack Completed :) Earlier than expected")
```